c/src/demo_masq_ds.c

```diff
@@ -0,0 +1,199 @@
+/******************************************************************************
+ *
+ * Quantropi CONFIDENTIAL
+ * ----------------------
+ *
+ * This file is proprietary code and subject to the terms and conditions
+ * defined in the file 'LICENSE', which is part of this source code package.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Quantropi Incorporated and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Quantropi Incorporated and its suppliers and
+ * may be covered by Canadian, U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Quantropi Incorporated.
+ *
+ *****************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include "../lib/masq_ds.h"
+#include "../lib/qispace_pqrnd.h"
+
+/* 
+ * This demo displays one example of the MASQ DS API, specifically MPPK level 5, 
+ * to sign a message between two parties(Alice & Bob) 
+ *
+ * In this case, Alice wants to send a secret message to Bob. 
+ * So, Alice begins by generating a public-private keypair.
+ * Alice uses its own private key and generate the signature. 
+ * Alice sends Bob, the public key and the signature. 
+ * Bob then uses Alice's public key to verify the message and signature. 
+ * Thus, both Alice and Bob arrive at the same message.
+ */
+
+unsigned char seed_orig[1024];
+int32_t seed_len = 32;  
+
+
+/* 
+* Initialize/set seed for any random number generator used
+* We recommand to call QiSpace API (i.e. QE) to get seed 
+* if no good quantum safe level entropy source
+*/
+int32_t seed_init()
+{
+    /* here we just use std lib rand to set seed for demo purpose */
+    srand(time(0));
+    for(int j = 0; j < seed_len; j++){
+            seed_orig[j] = rand() & 0xff;
+    }
+    return 1;
+}
+
+/* Customize the random callback function below according to specific needs. 
+ * Otherwise, default random CF is Quantropi 
+ */
+int32_t rand_cf(MASQ_RAND_handle rand_handle, int32_t rand_length,  uint8_t *rand_num){
+    
+    if (rand_num == NULL) return 0;
+    
+    if (rand_handle == NULL) 
+    {
+        for (int i = 0; i < rand_length; i++)
+        {
+            rand_num[i] = rand() & 0xff;
+        }
+    }
+    return 1;
+}
+
+int32_t rand_seed_cf(MASQ_RAND_handle rand_handle, uint8_t *seed, int32_t seed_length)
+{
+    if (seed == NULL) return 0;
+
+    if(rand_handle == NULL){
+        seed = seed_orig;
```

Comment:
unsigned int sd = 0;
for (int i=0; i< seed_length; i++)
   sd ^= seed[i];
srand(sd);